autogen/oai/aiinference.py

```diff
@@ -0,0 +1,231 @@
+from __future__ import annotations
+
+import json
+import logging
+import os
+import time
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import requests
+from openai.types.chat import ChatCompletion, ChatCompletionMessage
+from openai.types.chat.chat_completion import Choice
+from openai.types.completion_usage import CompletionUsage
+
+from autogen.oai.client_utils import validate_parameter
+
+logger = logging.getLogger(__name__)
+
+
+class AzureAIInferenceClient:
+    """Azure AI Inference Client
+
+    This class provides an interface to interact with Azure AI Inference API for natural language processing tasks.
+    It supports various language models and handles API requests, response processing, and error handling.
+
+    Key Features:
+    1. Supports multiple AI models provided by Azure AI Inference.
+    2. Handles authentication using API keys.
+    3. Provides methods for creating chat completions.
+    4. Processes and formats API responses into standardized ChatCompletion objects.
+    5. Implements rate limiting and error handling for robust API interactions.
+    6. Calculates usage statistics and estimated costs for API calls.
+
+    Usage:
+    - Initialize the client with the desired model and API key.
+    - Use the 'create' method to generate chat completions.
+    - Retrieve messages and usage information from the responses.
+
+    Note: Ensure that the AZURE_API_KEY is set in the environment variables or provided during initialization.
+
+    # Example usage
+    if __name__ == "__main__":
+    import os
+    import autogen
+
+    config_list = [
+        {
+            "model": "gpt-4o",
+            "api_key": os.getenv("AZURE_API_KEY"),
+        }
+    ]
+
+    assistant = autogen.AssistantAgent(
+        "assistant",
+        llm_config={"config_list": config_list, "cache_seed": 42},
+    )
+
+    human = autogen.UserProxyAgent(
+        "human",
+        human_input_mode="TERMINATE",
+        max_consecutive_auto_reply=10,
+        code_execution_config={"work_dir": "coding"},
+        llm_config={"config_list": config_list, "cache_seed": 42},
+    )
+
+    human.initiate_chat(
+        assistant,
+        message="Would I be better off deploying multiple models on cloud or at home?",
+    )
+    """
+
+    SUPPORTED_MODELS = [
```

Comment:
Do we need to check this from client side? Can we just let the server to check it.

Reply:
it could be a good idea to keep it server side because the updates are frequent , is there any interest in this ? on my side , sure , i could always send it in for review if it's a good idea :-)